lms/views/api/blackboard/files.py

```diff
@@ -0,0 +1,50 @@
+"""Proxy API views for files-related Blackboard API endpoints."""
+from pyramid.view import view_config, view_defaults
+
+from lms.views import helpers
+
+
+@view_defaults(permission="api", renderer="json")
+class BlackboardFilesAPIViews:
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(request_method="GET", route_name="blackboard_api.courses.files.list")
+    def list_files(self):
+        """Return the list of files in the given course."""
+        self.request.find_service(name="oauth2_token").get()
+
+        # Return a temporary hard-coded list of files.
+        return [
+            {
+                "id": "blackboard://content-resource/123",
+                "display_name": "Fake Blackboard File 1",
+                "updated_at": "2020-06-10T16:49:19Z",
+            },
+            {
+                "id": "blackboard://content-resource/456",
+                "display_name": "Fake Blackboard File 2",
+                "updated_at": "2020-06-10T16:48:53Z",
+            },
+            {
+                "id": "blackboard://content-resource/789",
+                "display_name": "Fake Blackboard File 3",
+                "updated_at": "2020-08-03T14:06:00Z",
+            },
+        ]
+
+    @view_config(request_method="GET", route_name="blackboard_api.files.via_url")
+    def via_url(self):
+        """Return the Via URL for annotating the given Blackboard file."""
+        self.request.find_service(name="oauth2_token").get()
```

Comment:
Duplicating the `get()` that raises `NoOAUth2Token`. Putting this in the `__init__()` just makes the tests awkward to write, and tangling views together with shared `__init__()` logic is usually a bad idea. This is very temporary code anyway